### PR TITLE
added 'Reference - Events' documentation

### PR DESCRIPTION
**Pull request by eric-chau**
Referenced issue: [backbee/backbee-standard #649](https://github.com/backbee/backbee-standard/issues/649)

ping @mickaelsteinberg.

### CODE REVIEW

**Comment by crouillon** [comment]
:+1: thx @eric-chau

File: doc/developer-documentation/references/events.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        - [BackBee\ClassContent\Listener\RevisionListener, onFlushContent]
+        - [BackBee\ClassContent\Listener\IndexationListener, onFlushContent]
+        - [@cache.listener, onFlushContent]
+        - [BackBee\Event\Listener\RewritingListener, onFlushContent]
+        - [BackBee\MetaData\Listener\MetaDataListener, onFlushContent, -100]
+```
+
+### Event name (e.g. `classcontent.onflush`)
+
+When you want to register an event listener, the first step is to define which event you want to listen. In the sample above, we added listener for the `classcontent.onflush` event.
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/events.md]: 2 minor changes to add : 
When you want to register an event listener, the first step is to define which event you want to listen **to**. In the sample above, we added **a** listener **to** the `classcontent.onflush` event.

File: doc/developer-documentation/references/events.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+- In bundle: `__BUNDLE_ROOT_DIR__/Config/events.yml`
+- In BackBee core: `__BACKBEE_ROOT_DIR__/Config/events.yml`
+- In BackBee Standard Edition: `__PROJECT_ROOT_DIR__/repository/Config/events.yml`
+
+If you want to load your events in a custom way, you should take a look to [the bundle component documentation](/developer-documentation/components/bundle/).
+
+## Configuration
+
+```
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/events.md]: add **yaml**

File: doc/developer-documentation/references/events.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        - [BackBee\MetaData\Listener\MetaDataListener, onFlushContent, -100]
+```
+
+### Event name (e.g. `classcontent.onflush`)
+
+When you want to register an event listener, the first step is to define which event you want to listen. In the sample above, we added listener for the `classcontent.onflush` event.
+
+### Callbacks
+
+Your event listener callback must be a callable array. It must contain at least 2 elements. The first one must be a fully-qualified class name (e.g. `BackBee\ClassContent\Listener\ClassContentListener`) and the second must be the method to call (e.g. `onFlushContent`). Note that BackBee does not instantiate the event listener class. It's invoked statically.
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/events.md]: I'm writing it with fewer sentences, what do you think ?

An event listener callback is an callable array containing at least 2 elements : 

* A fully-qualified class name (e.g. `BackBee\ClassContent\Listener\ClassContentListener`)
* The method to call (e.g. `onFlushContent`)

Note that Backbee statically invokes the event listener. No need to instantiate it.

File: doc/developer-documentation/references/events.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+If for some reasons you need to do some stuff on your event listener before it's ready to handle events, you can pass the identifier of a service from dependency injection container (e.g. `@cache.listener` - don't forget to prefix the service identifier by `@`).
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/events.md]: If you need to apply a specific behavior **before** your event listener is actually handling the event, you can specify a service identifier from the Dependency Injection Container (e.g. `@cache.listener` - remember to target the service identifier by using the `@` character).

+ Add an example below in the **Basic Usage** section (maybe rename from Basic Usage to Usage)

File: doc/developer-documentation/references/events.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+### Priority
+
+BackBee event dispatcher extends Symfony's one so you can manage your event listener priority too. By default, every event listener has the priority `0` (the highest the priority, the earlier a listener is executed). When events listeners listen to same event and have same priority, the declaration order determines which one is called first.
```
Comment (mickaelsteinberg) [inline doc/developer-documentation/references/events.md]: You can leverage your event listener's priority using Symfony's Event Dispatcher system, which BackBee extends. 

- Event listener's default priority is set to `0`
- The highest your priority, the earliest your event is triggered
- When two events listeners have the same priority, their declaration in the execution pile determine which one is called first

**Comment by mickaelsteinberg** [comment]
@eric-chau here is some ideas that could improve the reading. If you feel like integrate them, feel free to do so ;)
Thanks

**Comment by eric-chau** [comment]
@mickaelsteinberg thank you for your feedbacks! I will surely use them to improve my documentation!

**Comment by eric-chau** [comment]
Ok I applied all suggestion you did. Thank you @mickaelsteinberg!

**Comment by mickaelsteinberg** [comment]
👍

**Pull request merged**